src/app/api/search/route.ts

```diff
@@ -17,44 +19,82 @@ const fuzzySearch = (videos: TSearchedVideos[], searchQuery: string) => {
     minMatchCharLength: 3,
     keys: ['title'],
   }).search(searchQuery);
-
   return searchedVideos.map((video) => video.item);
 };
 
+function isAdmin(email: string): boolean {
+  const adminEmails = process.env.ADMINS?.split(',') || [];
+  return adminEmails.includes(email);
+}
+
 export async function GET(request: NextRequest) {
+  const session = await getServerSession(authOptions);
+  if (!session || !session.user || !session.user.email) {
+    return NextResponse.json({ error: 'Unauthorized' }, { status: 401 });
+  }
+
   const { searchParams } = new URL(request.url);
   const searchQuery = searchParams.get('q');
 
   if (searchQuery && searchQuery.length > 2) {
-    const value: TSearchedVideos[] = await cache.get(
-      'getAllVideosForSearch',
-      [],
-    );
+    const userIsAdmin = isAdmin(session.user.email);
+    const cacheKey = `getAllVideosForSearch:${session.user.id}:${userIsAdmin ? 'admin' : 'user'}`;
```

Comment:
Basically my concern is the use of cache is almost useless now. Mostly because we are caching per userid which will be a lot. Can u think of some hack or some sneaky way to prevent this.

Things that can work. Fetech all the data but only allow for searching not for vewing the content like playing the video or interacting with the comment section